app/controllers/LocationController.scala

```diff
@@ -48,4 +51,60 @@ class LocationController @Inject()(locationDao: LocationDao,
     }
   })
 
+  def postHistory(): Action[AnyContent] = securedAsync(Driver.INSTANCE :: Nil, Action.async {
+    request: Request[AnyContent] =>
+      logger.info("Received postponed location update request")
+      val body = request.body.asText
+      body match {
+        case Some(bodyString) =>
+          logger.info(bodyString)
+          val locations = Json.fromJson(bodyString, classOf[List[common.OffsetedLocationUpdate]])
+          currentUser(request).flatMap {
+            case Some(user) =>
+              val updated = locationService.postOfflineHistory(locations, user.getId)
+              updated.map {
+                res => Ok(res.toString)
+              }
+            case None =>
+              logger.error(s"User session was verified, but now no user found, THIS SHOULD NEVER HAPPEN")
+              Future.successful(InternalServerError(Json.toJson(
+                new common.InternalServerError("No user found for existing session, this should never happen"))))
+          }
+        case None =>
+          logger.info("Empty location update request, returning 401")
+          Future.successful(BadRequest(Json.toJson(
+            new BadRequest("Empty body not allowed for location update"))))
+      }
+  })
+
+  def getOwnHistory(from: Timestamp, to: Timestamp): Action[AnyContent] = securedAsync(Driver.INSTANCE :: Nil, Action.async {
```

Comment:
Maybe also add something like `getOwnHistory(number: Long)`?
Just a proposal

Reply:
you mean like last n locations?

Reply:
Yep

Reply:
Dunno, seems weird, can you come up with a usecase?

Reply:
I thought about printing this info by pages, but it should be another mechanism, forget it :C